cpuid.go

```diff
@@ -747,6 +753,32 @@ func (c *CPUInfo) cacheSize() {
 	return
 }
 
+type SGXSupport struct {
+	Available           bool
+	SGX1Supported       bool
+	SGX2Supported       bool
+	MaxEnclaveSizeNot64 int64
+	MaxEnclaveSize64    int64
+}
+
+func sgx(available bool) (rval SGXSupport) {
+	rval.Available = available
+
+	if !available {
+		return
+	}
+
+	a, b, c, d := cpuidex(0x12, 0)
+	rval.SGX1Supported = a&0x01 != 0
+	rval.SGX2Supported = a&0x02 != 0
+	rval.MaxEnclaveSizeNot64 = int64(math.Pow(2, float64(d&0xFF)))
```

Comment:
Instead of using float, I believe `1 << int64(d&0xFF)` will be 100% precise. Same for the following.